### PR TITLE
Diff - Add option to collapse files

### PR DESCRIPTION
**Pull request by EduardoCopat**
Implements #3165 

@FreHu @sbcgua

### CODE REVIEW

File: src/ui/zcl_abapgit_gui_page_diff.clas.abap [hunk 1]
```diff
@@ -1111,6 +1111,7 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_DIFF IMPLEMENTATION.
       ro_html->add( '<th class="num">new</th>' ).           "#EC NOTEXT
       ro_html->add( '<th class="mark"></th>' ).             "#EC NOTEXT
       ro_html->add( '<th>code</th>' ).                      "#EC NOTEXT
+      ro_html->add( '<th class="collapse" onclick="onDiffCollapse(event);">▼</th>' ). "#EC NOTEXT
```
Comment (EduardoCopat) [inline src/ui/zcl_abapgit_gui_page_diff.clas.abap]: Any idea how could I solve this abaplint issue?

**Contains non 7 bit ascii character
7bit_ascii**

Should I try to use an ASCII character ou replace it with an icon?

Comment (larshp) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap]: suggest: add "zcl_abapgit_gui_page_diff" to exclude in abaplint.json, under the rule

Comment (larshp) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap]: yea, moving it to the JS file might also work

Comment (EduardoCopat) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap]: I added the exception. Thanks!

**Comment by FreHu** [comment]
A few ideas:

I think the ideal location for the collapse button would be the left side, the screen is wide and there is nothing to click on the right. 

image
vs
image

It could be placed left the added/removed/modified counters, there is more space for it there, you could have a more noticeable icon.

When I collapse everything, I would like to have an overview of my staged changes. 
- add expand/collapse all options to the top menu
--------------------------------
Collapsing of partially staged files

image

In the first file, I have no lines selected, but in the second one I have one of two lines selected. They both make it look like nothing is selected. 
- There could be an extra line shown in collapsed mode which would summarize the user's selection ("1 of 2 lines staged")
- Different background colors for `nothing staged`/`something staged`/`everything staged`

**Comment by EduardoCopat** [comment]
@FreHu, thanks for the ideas! I will work on them.

I think we could follow how Github collapses:

image

image

They use the collapse button close to the counters.

**Comment by larshp** [comment]
@FreHu does the change look okay from your point of view?

**Comment by FreHu** [comment]
The collapsing works, but I'd merge it after arrows are moved to the left. The rest can go in different PRs.

**Comment by EduardoCopat** [comment]
ezgif com-video-to-gif

- I added a nice icon (chevron) for collapse/expand. Similar to github.

As mentioned, we can create issues for the other comments.

**Comment by sbcgua** [comment]
Heh, I'm too late everything is already reviewed. :)
From my side no suggestions, nice and clean code, cool and useful feature. :+1:

File: src/ui/zcl_abapgit_gui_page_diff.clas.abap (hunk 1 above)
Comment (sbcgua) [reply src/ui/zcl_abapgit_gui_page_diff.clas.abap]: Just FYI: there are unicode symbols escaping in html. This one would be `&#x25bc;` I guess

image

But chevron icons look better anyway

**Comment by larshp** [comment]
thanks guys 👍

**Pull request merged**